snowdrop-block/src/Snowdrop/Block/Chain.hs

```diff
@@ -9,19 +10,22 @@ module Snowdrop.Block.Chain
 import           Universum
 
 import           Snowdrop.Block.Configuration (BlkConfiguration (..), getCurrentBlockRef)
+import           Snowdrop.Block.State (BlundComponent)
 import           Snowdrop.Block.StateConfiguration (BlkStateConfiguration (..))
 import           Snowdrop.Block.Types (Block (..), BlockHeader, BlockRef, BlockUndo, Blund (..),
                                        CurrentBlockRef (..), PrevBlockRef (..), RawBlund,
                                        RawPayload)
+import           Snowdrop.Core (ChgAccum, ERoComp, QueryERo, queryOne)
 import           Snowdrop.Util (NewestFirst (..), OldestFirst (..), newestFirstFContainer,
                                 toOldestFirst)
 
 -- | Load up to @maxDepth@ blocks from the currently adopted block sequence.
 iterateChain
-    :: forall blkType m .
-    ( Monad m
+    :: forall blkType conf xs m .
```

Comment:
Looks like functions `iterateChain`, `wholeChain`, `forkDepthChain`, `nDepthChain`, `nDepthChainNE` should take `BlkConfiguration` as parameter, not `BlkStateConfiguration` (which they do not use)

Reply:
All mentioned functions depends on `iterateChain` which is using `bscGetTip` from `BlkStateConfiguration`.

Reply:
Alright :)